tests/sqllogictest/src/lib.rs

```diff
@@ -28,12 +27,7 @@ pub async fn test_mem(name: &str) {
 
 pub async fn test_disk(name: &str) {
     init_logger();
-    // still need a temp_dir to write manifest
-    let temp_dir = tempdir().unwrap();
```

Comment:
So now testing with the secondary storage will not actually touch files?

Reply:
Yes, I've manually removed all disk access. I think we will eventually have a object store abstraction in the future.

Reply:
It sounds unreasonable to test a disk backend without actually accessing files.

Reply:
Actually, the reason is that, we only test whether the disk encoder / decoder can correctly read the file format, without actually testing things like "fsync", especially for unit tests like sqllogictest.

Reply:
I still feel that sqllogictest is a kind of e2e test and it's better to cover the real execution path as much as possible. 🤔

Reply:
If we can abstract file operations into a single trait, then all paths are still covered, except the on-disk implementation of ObjectStore or FileSystem trait. Indeed RisingLight currently do not rely much on the filesystem magics and properties, so I don't think it will be necessary to cover that paths.